### ISSUE TITLE
Don't print info messages to STDOUT

### ISSUE THREAD

**Issue by huyderman** [manuelbua/gitver#13]
When executing this snippet in a fresh project with tags:

``` sh
#/usr/bin/env sh
gitver init > /dev/null
gitver current
```

I get the following output to STDOUT:

``` sh
(wrote default configuration file "/home/huyderman/gitver/.gitver/config"
0.3.0-RC1.11+4e5bfdc
```

To avoid the info-message in STDOUT, I would have to do something the following to ensure everything is properly initialized:

```
#/usr/bin/env sh
gitver init > /dev/null
gitver > /dev/null
gitver current
```

The info message makes the output unpredictable and tricky to parse in scripts, etc. This could be avoided by outputting info messages to STDERR, or having a `--silent` command line flag or similar.

**Comment by manuelbua** [comment]
Thanks for the feedback!
Mmm as far as i know, console messages shall be output to the STDOUT and error messages to STDERR and i agree this is currently _not_ the case: if you meant to have gitver use std streams properly, then sure that should be done, but i disagree in mixing stdout with stderr, if i understood you correctly.

I'm also missing the `--silent/--quiet` flag and that will be added, thanks!

I don't understand why you do `gitver > /dev/null` and `gitver current` after  the `gitver init > /dev/null`: can you clear this up?

> #/usr/bin/env sh
> gitver init > /dev/null
> gitver current

You shouldn't really get any output by redirecting to /dev/null, this i've to investigate more.

Note that, as it's being done in `git init`, the `gitver init` command output is needed to ensure the user has done everything properly, i.e. things such as the `.gitver` directory in the `.gitignore` file have to be taken care properly for `gitver` to function correctly and the warning message serves this purpose.

Instead, if you meant to script the gitver initialization and perform that step 'later in the script' because you already know how it works, then the `--quiet` flag will help with that for stdout messages, but still error messages shouldn't  be suppressed at all.

It looks like the common convention for warning messages is to output them to `stderr` and i'll probably go with that, this means you'll need to suppress stderr via the `2>/dev/null` redirection.

So to recap, i'm all for using streams properly and adding a quiet flag, but i won't mix normal output with errors, but please tell me more, i could be misunderstanding your reasoning.

**Comment by manuelbua** [comment]
Commit 4944feb7 adds the "--quiet" and "--quiet-errors" feature: you can either use these flags or just the plain old pipe redirection, such as `1>/dev/null`, `2>/dev/null`, `&>/dev/null`, or any other combination.

Previous commits fix std streams usage, hope this fix your issues now, thanks for bringing this up and helping in make it a better tool!

**Comment by huyderman** [comment]
I realize I was a bit unclear, I'll try explaining again. 

The posted snippet was from a script to get the version from a freshly checked out repository (e.g. pre-`gitver init`) by capturing stdout. Therefore `gitver init` was run in the script to initialize the repository, and the output suppressed by piping to `/dev/null`. This would leave only the output from `gitver current`. In hindsight, including the snippet just confuses the matter.

The problem was that on first time run, `gitver current` would print `(wrote default configuration file "/home/huyderman/gitver/.gitver/config"` to stdout before printing the version. If you were piping the output to somewhere that just expected a version string (e.g. `gitver current | sh myscript.sh`), this would probably cause something to break. By being able to ensure that only actual program output is printed to stdout, `gitver current` and other commands becomes more predictable and pipe-friendly. Being able to suppress info-messages and warnings with a command line flag solves this.

Many command line utilities print only program output to stdout, while printing info-messages and warnings to stderr. By doing this, you could pipe or capture stdout, while still displaying or logging info messages from stderr.

**Comment by manuelbua** [comment]
Ahhh, now i got it, thanks for the better explanation!
You know, English is also not my native language, so i probably miss it sometimes, please bear with me :)

Now that i've understood your point i've reasoned about it: i really should be creating the default configuration file at `gitver init` time and **not** at first launch, this should be more predictable and git-alike.

Also, the `.gitignore` check should really be moved to a `check` command instead, i don't even like the one-command multiple-things (check .gitignore + get version), but i thought it would be good for new-comers to have `gitver` to check for consistency when run.

I'm going to enter some issues in the Issues section, then this evening i'm going to work on this and update this issue later on for you to test: thank you very much for taking the time to elaborate on this, i really appreciate your feedback @huyderman!

> Many command line utilities print only program output to stdout, while printing info-messages and warnings to stderr. By doing this, you could pipe or capture stdout, while still displaying or logging info messages from stderr.

This i missed completely from your previous post :+1: 

I suspect this will see me busy for some time, but hopefully i should be able to do that before returning full-time on the game i'm working on, in fact i started this tool because i needed a way to keep track of version strings between the client and the server, i never thought it would be a so well-received piece of software, that was unexpected!

**Comment by huyderman** [comment]
You're welcome. :smile: 
I've been looking for a good way to handle version strings myself when I found a link to this project on reddit. I like the approach `gitver` uses, I think it'll fit my workflow well.
